### ISSUE TITLE
`vitest/no-conditional-tests` false positive

### ISSUE THREAD

**Issue by jakubmazanec** [veritem/eslint-plugin-vitest#358]
To simplify, I believe `vitest/no-conditional-tests` rule should be triggered if `test` is under an `if`, and `vitest/no-conditional-in-test` should be triggered if there's an `if` under a `test` (similar to https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-in-test.md).

But when linting this code (with both rules turned on):

```
test('test1', () => {
  if (Math.random() > 0) { // `vitest/no-conditional-in-test` error should be here
    doSomething();
  }

  expect(true).toBeTruthy();
});

if (Math.random() > 0) { // `vitest/no-conditional-tests` error should be here
  test('test2', () => {
    expect(true).toBeTruthy();
  });
}
```

the result is different:

1) there are two errors reported in the first test, but only `vitest/no-conditional-in-test` should error
2) there is no error reported in the second test, but `vitest/no-conditional-tests` should error.

**Comment by veritem** [comment]
a pr is welcome
